File: ios/TmuxWebClient/Storage/ConnectionSettingsStore.swift
import Foundation

/// Server URL is not a secret so it lives in UserDefaults; the token does
/// (see KeychainStore.swift). Mirrors what the browser UI keeps in
/// localStorage, but split across two stores by sensitivity.
struct ConnectionSettings: Equatable {
    var baseURL: URL
    var token: String
}

final class ConnectionSettingsStore: ObservableObject {
    private static let baseURLKey = "tmuxweb.baseURL"

    @Published private(set) var current: ConnectionSettings?

    init() {
        current = Self.load()
    }

    func save(baseURL: URL, token: String) throws {
        UserDefaults.standard.set(baseURL.absoluteString, forKey: Self.baseURLKey)
        try KeychainStore.saveToken(token)
        current = ConnectionSettings(baseURL: baseURL, token: token)
    }

    func clear() {
        UserDefaults.standard.removeObject(forKey: Self.baseURLKey)
        KeychainStore.deleteToken()
        current = nil
    }

    private static func load() -> ConnectionSettings? {
        guard
            let urlString = UserDefaults.standard.string(forKey: baseURLKey),
            let baseURL = URL(string: urlString),
            let token = KeychainStore.loadToken()
        else {
            return nil
        }
        return ConnectionSettings(baseURL: baseURL, token: token)
    }
}

File: kmp/composeApp/src/commonMain/kotlin/com/tanyudii/tmuxweb/domain/repository/ConnectionSettingsStore.kt
package com.tanyudii.tmuxweb.domain.repository

import com.tanyudii.tmuxweb.data.local.BaseUrlStore
import com.tanyudii.tmuxweb.data.local.TokenStore
import com.tanyudii.tmuxweb.domain.model.ConnectionSettings

/** Ports ConnectionSettingsStore.swift's `save`/`clear`/`load` — same repository-interface pattern as Phase 2. */
interface ConnectionSettingsStore {
    suspend fun load(): ConnectionSettings?
    suspend fun save(baseUrl: String, token: String)
    suspend fun clear()
}

class DefaultConnectionSettingsStore(
    private val tokenStore: TokenStore,
    private val baseUrlStore: BaseUrlStore,
) : ConnectionSettingsStore {
    override suspend fun load(): ConnectionSettings? {
        val baseUrl = baseUrlStore.loadBaseUrl() ?: return null
        val token = tokenStore.loadToken() ?: return null
        return ConnectionSettings(baseUrl, token)
    }

    override suspend fun save(baseUrl: String, token: String) {
        baseUrlStore.saveBaseUrl(baseUrl)
        tokenStore.saveToken(token)
    }

    override suspend fun clear() {
        baseUrlStore.deleteBaseUrl()
        tokenStore.deleteToken()
    }
}
